### PR TITLE
updated model persistor docs and aws persistor

### LINKED ISSUE #5736: Update azure model persistor
- follow up issue for https://github.com/RasaHQ/rasa/pull/5719

Todos:
- [ ] update version of `azure-storage-blob` dependency
- [ ] update & test code of `azure` persistor

> Microsoft released a new version with breaking changes in late 2019. Rasa uses the legacy version [...]

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- updated model persistor docs
- update AWS persistor to use the latest package fixes #5736
- make sure to use `pip3` instead of `pip`

**Status (please check what you already did)**:
- [x] added some tests for the functionality (not possible for )
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/model-storage.mdx [hunk 1]
```diff
@@ -2,156 +2,214 @@
 id: model-storage
 sidebar_label: Model Storage
 title: Model Storage
+abstract: |
+  After you've trained a model for your assistant, you can use it to run your
+  bot.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:7]: This is a bit confusing, as it doesn't really describe what model storage does

File: docs/docs/model-storage.mdx [hunk 2]
```diff
@@ -2,156 +2,214 @@
...
+  header of the request matches the model you want your server to return.
+
+Rasa Open Source uses the `If-None-Match` and `ETag` headers fro caching. Setting
+the headers will avoid re-downloading the same model over and over, saving
+bandwidth and compute resources.
 
-## Fetching Models from Remote Storage
+An example request Rasa might make to your model server looks like this:
+
+## Load Model from Cloud
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:104]: I thought we were going with `"ing"` style headers?

File: docs/docs/model-storage.mdx [hunk 3]
```diff
@@ -2,156 +2,214 @@
...
 
-Rasa can load your model in three different ways:
+You can load your trained model
+in three different ways:
 
-1. Fetch the model from a server (see [Fetching Models from a Server](./model-storage.mdx#server-fetch-from-server)), or
+1. Load the model from your local disk (see [Load Model from Disk](./model-storage.mdx#load-model-from-disk))
 
-2. Fetch the model from a remote storage (see [Remote Storage](./model-storage.mdx#server-fetch-from-remote-storage)).
+2. Fetch the model from your own http server (see [Load Model from Server](./model-storage.mdx#load-model-from-server))
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:15]:
```suggestion
2. Fetch the model from your own HTTP server (see [Load Model from Server](./model-storage.mdx#load-model-from-server))
```

File: docs/docs/model-storage.mdx [hunk 4]
```diff
@@ -2,156 +2,214 @@
...
 
-3. Load the model specified via `-m` from your local storage system.
+3. Fetch the model from cloud storage like S3 (see [Load Model from Cloud](./model-storage.mdx#load-model-from-cloud))
 
-Rasa tries to load a model in the above mentioned order, i.e. it only tries to load your model from your local
-storage system if no model server and no remote storage were configured.
+Loading the model from your local disk will be used as a fallback if you
+don't specify how a model should be loaded.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:20]: Can we rephrase this to make it sound more like a default than a fallback (this here feels like it implies you need to specify)

Comment (erohmensing) [reply docs/docs/model-storage.mdx:20]: the intro to the next section is much clearer about this!

File: docs/docs/model-storage.mdx [hunk 5]
```diff
@@ -2,156 +2,214 @@
...
+
+```bash
+rasa run --model models/
+```
+
+Rasa will check all the models in that directory and load the one which was trained
+most recently.
+
+Rasa will assume `models/` to be the default model directory and
+will use it in case no `--model` argument is specified. The two following calls
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:42]:
```suggestion
If you don't specify a `--model` argument, Rasa Open Source will look for models in the  `models/` directory. The two following calls
```

File: docs/docs/model-storage.mdx [hunk 6]
```diff
@@ -2,156 +2,214 @@
...
+
+If you want to load the latest model in a directory, you can specify
+a directory instead of a file:
+
+```bash
+rasa run --model models/
+```
+
+Rasa will check all the models in that directory and load the one which was trained
+most recently.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:39]:
```suggestion
Rasa Open Source will check all the models in that directory and load the one that was trained
most recently.
```

File: docs/docs/model-storage.mdx [hunk 7]
```diff
@@ -2,156 +2,214 @@
...
-reguarly query this URL for zipped models.
+by adding it to your `endpoints.yml`:
 
 ```yaml-rasa title="endpoints.yml"
 models:
-  url: http://my-server.com/models/default@latest
+  url: http://my-server.com/models/default
   wait_time_between_pulls: 10   # In seconds, optional, default: 100
 ```
+The server will query `url` for a zipped model every `wait_time_between_pulls`
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:67]:
```suggestion
The server will query the `url` for a zipped model every `wait_time_between_pulls`
```

File: docs/docs/model-storage.mdx [hunk 8]
```diff
@@ -2,156 +2,214 @@
...
-If you want to pull the model just once from the server, set
-`wait_time_between_pulls` to `None`.
-
-:::
-
-:::note
-Your model server must provide zipped Rasa models, and have
-`{"ETag": <model_hash_string>}` as one of its headers. Rasa will
-only download a new model if this model hash has changed.
+If you want to pull the model just once from the server, you can set the time
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:70]:
```suggestion
If you want to pull the model only when starting up the server, you can set the time
```

File: docs/docs/model-storage.mdx [hunk 9]
```diff
@@ -2,156 +2,214 @@
...
+  url: http://my-server.com/models/default
+  wait_time_between_pulls: null  # fetches model only once
+```
 
-Rasa sends requests to your model server with an `If-None-Match`
-header that contains the current model hash. If your model server can
-provide a model with a different hash from the one you sent, it should send it
-in as a zip file with an `ETag` header containing the new hash. If not, Rasa
-expects an empty response with a `204` or `304` status code.
+#### Configuration of your server
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:79]:
```suggestion
#### Configuration of Your Server
```

Comment (erohmensing) [reply docs/docs/model-storage.mdx:79]: what about "How to Configure Your Server" ? would be more straightforward i think

File: docs/docs/model-storage.mdx [hunk 10]
```diff
@@ -2,156 +2,214 @@
...
 
-An example request Rasa might make to your model server looks like this:
+Rasa Open Source will send a `GET` request to the URL you specified in the
+`endpoints.yml`, e.g. in the above examples this would be
+`http://my-server.com/models/default`. You can use any URL.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:83]:
```suggestion
`endpoints.yml`, e.g. `http://my-server.com/models/default` in the above examples.
You can use any URL.
```

File: docs/docs/model-storage.mdx [hunk 11]
```diff
@@ -2,156 +2,214 @@
...
-curl --header "If-None-Match: d41d8cd98f00b204e9800998ecf8427e" http://my-server.com/models/default@latest
+curl --header "If-None-Match: d41d8cd98f00b204e9800998ecf8427e" http://my-server.com/models/default
 ```
 
-<a aria-hidden="true" tabIndex="-1" className="anchor enhancedAnchor" id="server-fetch-from-remote-storage"></a>
+The response of your server to this `GET` request should be one of these:
+- return a status code of `200`, a zipped Rasa Model and set the `ETag` header in
+  the response to the hash of the model.
+- return a status code of `304` and an empty response if the `If-None-Match`
+  header of the request matches the model you want your server to return.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:96]:
```suggestion
- a status code of `200`, a zipped Rasa Model and set the `ETag` header in
  the response to the hash of the model.
- a status code of `304` and an empty response if the `If-None-Match`
  header of the request matches the model you want your server to return.
```

File: docs/docs/model-storage.mdx [hunk 12]
```diff
@@ -2,156 +2,214 @@
...
+
+Rasa Open Source uses the `If-None-Match` and `ETag` headers fro caching. Setting
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:98]:
```suggestion
Rasa Open Source uses the `If-None-Match` and `ETag` headers for caching. Setting
```

File: docs/docs/model-storage.mdx [hunk 13]
```diff
@@ -2,156 +2,214 @@
...
+the headers will avoid re-downloading the same model over and over, saving
+bandwidth and compute resources.
 
-## Fetching Models from Remote Storage
+An example request Rasa might make to your model server looks like this:
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:102]: ?

File: docs/docs/model-storage.mdx [hunk 14]
```diff
@@ -2,156 +2,214 @@
...
-The model will be downloaded from the remote storage system and stored in a temporary directory on your local storage system.
-Rasa supports using [S3](https://aws.amazon.com/s3/) ,
-[GCS](https://cloud.google.com/storage/) and [Azure Storage](https://azure.microsoft.com/services/storage/)
-to save your models, or you can write a custom module to connect to remote storage.
-
-Before you save your models to the cloud, they should be gzipped under the file name `{MODEL_NAME}.tar.gz`.
-The model should be stored in the root folder of the storage service.
-Currently, you are not able to manually specify the path on the cloud storage.
+The zipped model will be downloaded from cloud storage, unzipped and deployed.
+Rasa supports loading models from
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:113]:
```suggestion
The zipped model will be downloaded from cloud storage, unzipped, and deployed.
Rasa supports loading models from:
```

File: docs/docs/model-storage.mdx [hunk 15]
```diff
@@ -2,156 +2,214 @@
...
+- [Amazon S3](https://aws.amazon.com/s3/),
+- [Google Cloud Storage](https://cloud.google.com/storage/),
+- [Azure Storage](https://azure.microsoft.com/services/storage/) and
+- custom implementations for
+  [Other Remote Storages](./model-storage.mdx#other-remote-storages)
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:118]:
```suggestion
  [Other Remote Storages](./model-storage.mdx#other-remote-storages).
```

Also please add `Storages` to typoci

File: docs/docs/model-storage.mdx [hunk 16]
```diff
@@ -2,156 +2,214 @@
...
-### Azure Storage
+Google Cloud Storage (GCS) is supported using the `google-cloud-storage` package
+which you need to install as an additional dependency using `pip3`:
+```bash
+pip3 install google-cloud-storage
+```
 
-Azure is supported using the legacy `azure-storage-blob` package (v 2.1.0),
-which you can install with `pip install -I azure-storage-blob==2.1.0`.
+If you are running Rasa on a Google App Engine and Compute Engine, the auth
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:161]:
```suggestion
If you are running Rasa on Google App Engine or Compute Engine, the auth
```

File: docs/docs/model-storage.mdx [hunk 17]
```diff
@@ -2,156 +2,214 @@
...
 
-Start the Rasa server with `remote-storage` option set to `azure`.
+If you are running locally or on a machine outside of GAE or GCE you need to
+provide the authentication details to Rasa manually:
+1. Checkout the [GCS documentation](https://cloud.google.com/docs/authentication/getting-started#auth-cloud-implicit-python)
+   and follow the descriptions on "Creating a service account" and
+   "Setting the environment variable".
+2. After you completed the GCS instructions, you should have set an environment
+   variable called `GOOGLE_APPLICATION_CREDENTIALS` to the path of a service account
+   key file with access to your GCS.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:172]:
```suggestion
1. Check out the [GCS documentation](https://cloud.google.com/docs/authentication/getting-started#auth-cloud-implicit-python)
   and follow the descriptions on "Creating a service account" and
   "Setting the environment variable."
2. After you have completed the GCS instructions, you should have set an environment
   variable called `GOOGLE_APPLICATION_CREDENTIALS` to the path of a service account
   key file with access to your GCS.
```

File: docs/docs/model-storage.mdx [hunk 18]
```diff
@@ -2,156 +2,214 @@
...
 ```bash
-rasa run --remote-storage <your module>.<class name>
+rasa run --model 20190506-100418.tar.gz --remote-storage azure
 ```
 
-## Fetching Models from Local Storage
+### Other Remote Storages
+
+If you want to use any other Cloud Storage, you can provide your own python
+implementation of the `rasa.nlu.persistor.Persistor` class.
```
Comment (erohmensing) [inline docs/docs/model-storage.mdx:208]:
```suggestion
implementation of the [`rasa.nlu.persistor.Persistor`](/reference/rasa/nlu/persistor) class.
```

**Review by erohmensing — APPROVED** [review]
Added some nitpicks but they're all really small stuff. We need to figure out what we're calling Rasa and what we're calling Rasa Open Source in the docs as it's quite inconsistent 😄

File: docs/docs/model-storage.mdx (hunk 13 above)
Comment (tmbo) [reply docs/docs/model-storage.mdx:102]: just a leftover from the previous version, the example as already further up 🙂

File: docs/docs/model-storage.mdx (hunk 2 above)
Comment (tmbo) [reply docs/docs/model-storage.mdx:103]: mhm this was the shortest title that fit on a single line in the right-hand navbar so I went with it, do you think the `-ing` is important here?

Comment (erohmensing) [reply docs/docs/model-storage.mdx:99]: I think its easier to read/understand on first glance -- the length of the title is something we should look at again bc its currently wayy too short and not really feasible to shorten all the pages like this 🤔

**Comment by wochinge** [comment]
Was this force merged 👀  ?
Screenshot 2020-09-30 at 11 28 37
